webapp/src/js/services/db-sync.js

```diff
@@ -52,122 +46,110 @@ angular
-          // when we detect it's a 401
-          $log.error('Denied replicating ' + direction + ' remote server', err);
-        })
-        .on('error', function(err) {
-          $log.error('Error replicating ' + direction + ' remote server', err);
-        })
-        .on('complete', function(info) {
-          if (!info.ok && authenticationIssue(info.errors)) {
-            Session.navigateToLogin();
-          }
-        });
-    };
-
-    const replicateTo = () => {
-      const AUTH_FAILURE = {};
-      return Auth('can_edit')
-        // not authorized to replicate to server - that's ok. Silently skip replication.to
-        .catch(() => AUTH_FAILURE)
-        .then(err => {
-          if (err !== AUTH_FAILURE) {
-            return replicate('to');
+    ];
+
+    const replicate = function(direction) {
+      return direction.allowed()
+        .then(allowed => {
+          if (!allowed) {
+            // not authorized to replicate - that's ok, skip silently
+            return;
           }
+          const remote = DB({ remote: true });
+          return DB()
+            .replicate[direction.name](remote, direction.options)
+            .on('denied', function(err) {
+              // In theory this could be caused by 401s
+              // TODO: work out what `err` looks like and navigate to login
+              // when we detect it's a 401
+              $log.error(`Denied replicating ${direction.name} remote server`, err);
+            })
+            .on('error', function(err) {
+              $log.error(`Error replicating ${direction.name} remote server`, err);
+            })
+            .on('complete', function(info) {
+              const authError = !info.ok &&
+                info.errors &&
+                info.errors.some(error => error.status === 401);
+              if (authError) {
+                Session.navigateToLogin();
+              }
+            })
+            .then(() => {
+              return; // success
+            })
+            .catch(err => {
+              $log.error(`Error replicating ${direction.name} remote server`, err);
+              return direction.name;
+            });
         });
     };
 
-    const sendUpdateForDirectedReplication = (func, direction) => {
-      return func
-        .then(() => {
-          sendUpdate({
-            direction,
-            directedReplicationStatus: 'success',
-          });
-        })
-        .catch(error => {
-          sendUpdate({
-            direction,
-            directedReplicationStatus: 'failure',
-            error,
-          });
-          return error;
-        });
-    };
-
-    var syncMeta = function() {
-      var remote = DB({ meta: true, remote: true });
-      var local = DB({ meta: true });
-      local.sync(remote);
-    };
+    const getCurrentSeq = () => DB().info().then(info => info.update_seq + '');
+    const getLastSyncSeq = () => $window.localStorage.getItem(LAST_REPLICATED_SEQ_KEY);
```

Comment:
The const is calling it replication and the function is calling it sync. Requesting consistency.

Reply:
Fixed.